Debug-custom-ESLint-check-tests.md

```diff
@@ -0,0 +1,101 @@
+## Table of contents
+
+* [Introduction](#introduction)
+* [Print statements](#print-statements)
+* [Inspect the AST](#inspect-the-ast)
+
+## Introduction
+
+Especially for complex lint rules, it can be hard to figure out why a test failed. This page will give you some tips to help you debug your failing tests.
+
+## Print statements
+
+If you add a `console.log()` or `console.error()` statement to your custom ESLint rule code, the output will appear along with the test status output . For example, suppose we have a lint rule that raises an error every time `.click()` or `.sendKeys()` is called on an object instead of `action.click()` or `action.sendKeys()`, which we would like people to use instead. If we add `console.log()` statement to print `DEBUGGING found lint error` whenever our lint rule raises an error, we see this output:
+
+```text
+  e2e-action
+    valid
+      ✓ it('should test a feature', function() {
+        action.click("Element", elem);
+      });
+      ✓ it('should test a feature', function() {
+        action.sendKeys("Element", elem, "keys");
+      });
+      ✓ it('should test a feature', function() {
+        console.log(elem.click);
+      });
+      ✓ it('should test a feature', function() {
+        console.log(elem.sendKeys);
+      });
+    invalid
```

Comment:
We don't use this test format for eslint tests, you can see the format for eslint tests here(https://github.com/oppia/oppia/blob/develop/scripts/linters/custom_eslint_checks/rules/sorted-dependencies.spec.js)

Reply:
This is copied from the output I got when running `scripts/run_custom_eslint_tests.py`, so I'm pretty sure this is the correct test output format. Note that this is the test *output*, not the test definitions like in the file you linked to

Reply:
Ah-- I see.